src/libraries/System.Net.Security/src/System/Net/Security/TlsFrameHelper.cs

```diff
@@ -179,27 +179,51 @@ public static bool TryGetFrameHeader(ReadOnlySpan<byte> frame, ref TlsFrameHeade
             {
                 header.Type = (TlsContentType)frame[0];
 
-                if (frame.Length >= 3)
+                if (frame.Length > 4)
                 {
                     // SSLv3, TLS or later
                     if (frame[1] == 3)
```

Comment:
Where are all these magic numbers coming from?  Is it possible to copy/paste as comments the relevant portions of the RFC?

Reply:
Yes, we possibly can. In SSL3 and above fist 5 bytes is TLS frame header. In this particular case I can probably use `(frame.Length >= TlsFrameHelper.HeaderSize)`

The values are Type, Length and Version.